src/detect_entities.py

```diff
@@ -1,33 +1,41 @@
-from typing import List, TypedDict
+from typing import List
 import spacy
+from spacy.tokens import Span
 from src.entity_utils import MarkedEntity, is_entity_contained
 
-
 nlp = spacy.load("en_core_web_lg")
 
 
+def split_person_entity(entity: Span, source: str) -> List[MarkedEntity]:
+    split_entity: List[MarkedEntity] = []
+    ent_parts = entity.text.split(" ")
+    start_idx = entity[0].idx
+    for sub_ent in ent_parts:
+        split_entity.append(
+            {
+                "ent": sub_ent,
+                "type": "PERSON",
+                "start": start_idx,
+                "end": start_idx + len(sub_ent),
+                "in_source": is_entity_contained(sub_ent, source),
+            }
+        )
+        start_idx += len(sub_ent) + 1
```

Comment:
I guess the `+ 1` handles the fact that the multiple parts of a name can't overlap / are separated by whitespace? In the removed diff, it seems like we're always using spacy's `entity.idx` to get the starting index.

Why not continue to leverage spacy here? I feel like the less manual handling of indices, the better.

UPDATE: Ah, I think this is because we're taking a single multi-word span, i.e. `Daniel Levenson` and now handling it directly as a `List[str]`, is that right?

Reply:
Your last comment is correct - we now split by space instead of using spacy to split into sub entites.